miner/proposal_builder.go

```diff
@@ -405,13 +404,13 @@ func (pb *ProposalBuilder) handleLayer(ctx context.Context, layerID types.LayerI
 
 func (pb *ProposalBuilder) createProposalLoop(ctx context.Context) {
 	for {
+		next := pb.clock.GetCurrentLayer().Add(1)
 		select {
 		case <-pb.ctx.Done():
 			return
-
-		case layerID := <-pb.layerTimer:
+		case <-pb.clock.AwaitLayer(next):
 			lyrCtx := log.WithNewSessionID(ctx)
-			_ = pb.handleLayer(lyrCtx, layerID)
+			_ = pb.handleLayer(lyrCtx, next)
```

Comment:
Yeah, I added checks to make sure the same layer isn't processed twice.

I am not 100% sure how the layer subscription would have behaved in this case but this is how it behaves now:

- `pb.handleLayer` is never called twice on the same layer
- jumps backwards in time just cause the loop to wait again for the correct layer
- jumps forwards in time skip layers

This should correctly handle small time deviations backwards and any time deviations forwards. Large time deviations backwards (e.g. moving 100 layers back) will cause the `ProposalBuilder` to stop working until the layer it is expecting has arrived.

This might be incorrect, but I think would mimic the behaviour as it is with `Subscribe`